java/ql/src/semmle/code/java/dataflow/internal/ContainerFlow.qll

```diff
@@ -222,23 +240,38 @@ private predicate taintPreservingArgumentToQualifier(Method method, int arg) {
   method.(CollectionMethod).hasName("offer") and arg = 0
   or
   // java.util.Deque
+  // covered by Stack: push(E)
   method.(CollectionMethod).hasName(["addFirst", "addLast", "offerFirst", "offerLast"]) and arg = 0
   or
   // java.util.concurrent.BlockingQueue
+  // covered by Queue: offer(E, long, TimeUnit)
   method.(CollectionMethod).hasName("put") and arg = 0
   or
   // java.util.concurrent.TransferQueue
   method.(CollectionMethod).hasName(["transfer", "tryTransfer"]) and arg = 0
   or
   // java.util.concurrent.BlockingDeque
+  // covered by Deque: offerFirst(E, long, TimeUnit), offerLast(E, long, TimeUnit)
   method.(CollectionMethod).hasName(["putFirst", "putLast"]) and arg = 0
+  or
+  //java.util.Dictionary
+  method
+      .getDeclaringType()
+      .getSourceDeclaration()
+      .getASourceSupertype*()
+      .hasQualifiedName("java.util", "Dictionary") and
+  method.hasName("put") and
+  arg = 1
 }
 
 /**
  * Holds if `method` is a library method that returns tainted data if its
  * `arg`th argument is tainted.
  */
 private predicate taintPreservingArgumentToMethod(Method method, int arg) {
+  // java.util.Stack
+  method.(CollectionMethod).hasName("push") and arg = 0
```

Comment:
@aschackmull, isn't that even data flow then and not only tainting?

Reply:
I concur^^